### DISCUSSION TITLE
Feature request - API to return the raw speech probabilities

### DISCUSSION THREAD

**Discussion by ryanheise** [snakers4/silero-vad#274] [Q&A] [answered]
## 🚀 Feature

`get_speech_timestamps` may be useful function, but an API to get the raw speech probabilities is needed for more advanced scenarios.

## Motivation

As a very simple example, I may want to use an iterative approach to selecting the threshold parameters until finding the parameters that produce the desired number of segments per minute. For that, we don't want to keep running `get_speech_timestamps` multiple times to regenerate the speech probabilities, we'd want to generate the speech probabilities once and then do multiple passes on that.

But since that is just one very simplified example, I am not proposing a specific API just to implement that one specific example, but rather, direct access to the raw speech probs to allow the flexibility for apps to use the speech probs how they like.

## Pitch

Have torch hub return another function that gets the raw speech probabilities.

## Alternatives

The alternative is to create a fork of silero-vad.

**Comment by snakers4** [comment]
Hi, 

You can just look up how models are invoked in `get_speech_timestamps` and use them accordingly.

If you believe that your use case may be useful for a significant proportion of users, please do a PR.

Typically users do not want to mess with probabilities.

**Comment by adamnsandle** [comment, marked as answer]
You can use [`.audio_forward`](https://github.com/snakers4/silero-vad/blob/82d199ff22ae8897946ebb374d216cf6ce4f9aa3/utils_vad.py#L76) method for that purpose. Both models got this method in the latest v4 release.
```python3
wav = read_audio('files/en.wav')
raw_probs = model.audio_forward(wav, sr=16000, num_samples=512)
# tensor([[0.0948, 0.1472, 0.1674,  ..., 0.9034, 0.9971, 0.9988]])
```
